db/music.dbml

```diff
@@ -0,0 +1,299 @@
+Table submission {
+  submission_id uuid [pk, default: `gen_random_uuid()`]
+  participant_id int [not null, ref: > participant.participant_id]
+  bracket_id int [not null, ref: > bracket.bracket_id]
+  song_id int [not null, ref: > song.song_id]
+  order int [not null]
+  date_submitted timestamp [not null, default:`now()`]
+  active bool_int [not null]
+  is_dupe bool_int [not null]
+  title text [not null]
+  version text [null]
+  artist text [null]
+  collaborators text [null]
+  composer text [null]
+  link text [not null]
+  genre text [null]
+  emoji text [null]
+  comments text [null]
+
+  indexes {
+    (participant_id, bracket_id, song_id) [unique]
+  }
+}
+
+Table submission_history {
+  submission_history_id uuid [pk, default: `gen_random_uuid()`]
+  date_update timestamp [not null]
+  note_update text [not null, Note: 'dupe ineligible withdrawn "play-in loser"' ]
+  submission_id int [not null, ref: > submission.submission_id]
+  participant_id int [not null, ref: > participant.participant_id]
+  bracket_id int [not null, ref: > bracket.bracket_id]
+  song_id int [not null, ref: > song.song_id]
+  order int [not null]
+  date_submitted timestamp [not null, default:`now()`]
+  active bool_int [not null]
+  is_dupe bool_int [not null]
+  title text [not null]
+  version text [null]
+  artist text [null]
+  collaborators text [null]
+  composer text [null]
+  link text [not null]
+  genre text [null]
+  emoji text [null]
+  comments text [null]
+
+  indexes {
+    submission_id
+    participant_id
+    bracket_id
+    song_id
+  }
+
+  Note: 'composite unique index on submission_id, date_function (https://stackoverflow.com/a/26304364/2506158)'
```

Comment:
I don't think we need to worry about this, index on `submission_id` and on `date_update` sounds good (probably individually, but a composite index should be fine here too), but a unique index is probably asking for trouble.

Reply:
Maybe so. We've got a similar table at work and I always worry about it getting effed up. Never does, though.